chaos_genius/alerts/anomaly_alerts.py

```diff
@@ -297,67 +374,155 @@ def check_and_send_alert(self):
 
         Note: must only be called once on an instance.
         """
-        anomaly_data = self._get_anomalies()
+        anomaly_data = self._get_anomalies(
+            # consider day's offset only if it's a digest alert.
+            # when it's individual we want alert only for the last timestamp.
+            consider_days_offset=not self._to_send_individual()
+        )
```

Comment:
@Amatullah I wanted your thoughts on this. This happens only when the alert is run for the first time (so it's the first anomaly run after alert is set up).
If the alert was originally set to individual, we would store TriggeredAlerts only for the latest day (T-0 in case of hourly KPI). If the same alert was flipped to be a consolidated report, we would not get it in that day's and the next day's reports (it would first show up in a digest two days later).

A simple fix would be to always consider the days offset on the first run, for both individual and digest. We'll be adding extra rows of TriggeredAlerts that won't be used unless an individual alert is flipped to be a digest. This sounds like a reasonable tradeoff to me. @Amatullah what do you think?

Reply:
Sounds good. You can add the change.

Reply:
Decided to go with the fix after discussing with @Amatullah 

Fixed in 4fb3064702768f1f244853f82c515a3c6bb85031